R/bambu-extendAnnotations-utilityCombine.R

```diff
@@ -1,252 +1,312 @@
-            summarise_all(min, na.rm = TRUE)
-        end.unsplicedSum <- readClassSeTBL %>%
-            filter(confidenceType == "unsplicedNew") %>%
-            dplyr::select(end) %>%
-            mutate(index = overlapToCombined) %>%
-            group_by(index) %>%
-            summarise_all(max, na.rm = TRUE)
+combineUnsplicedTranscriptModels <- 
+    function(readClassList,  bpParameters, stranded, min.readCount, 
+             min.readFractionByGene, min.geneFDR, min.txFDR,
+             verbose){
+        start.ptm <- proc.time()
+        newUnsplicedSeList <- 
+            bplapply(seq_along(readClassList), function(sample_id)
+                extractNewUnsplicedRanges(readClassSe = readClassList[[sample_id]],
+                                          sample_id = sample_id), BPPARAM = bpParameters)
+        end.ptm <- proc.time()
+        if (verbose) message("extract new unspliced ranges object for all
+        samples in ", round((end.ptm - start.ptm)[3] / 60, 1)," mins.")
+        rangesList <- bplapply(newUnsplicedSeList, function(newUnsplicedSe){
+            rr <- unlist(rowRanges(newUnsplicedSe))
+            rr$row_id <- names(rr)
+            return(rr)
+        }, BPPARAM = bpParameters)
+        colDataNames <- unlist(lapply(newUnsplicedSeList, function(x) colnames(x)))
+        start.ptm <- proc.time()
+        combinedNewUnsplicedSe <- reduceUnsplicedRanges(rangesList, stranded)
+        end.ptm <- proc.time()
+        if (verbose) message("reduce new unspliced ranges object across all
+        samples in ", round((end.ptm - start.ptm)[3] / 60, 1)," mins.")
+        start.ptm <- proc.time()
+        combinedUnsplicedTibble <- 
+            makeUnsplicedTibble(combinedNewUnsplicedSe,newUnsplicedSeList, 
+                                colDataNames, min.readCount, min.readFractionByGene, min.geneFDR,
+                                min.txFDR, bpParameters)
+        end.ptm <- proc.time()
+        if (verbose) message("combine new unspliced tibble object across all
+        samples in ", round((end.ptm - start.ptm)[3] / 60, 1)," mins.")
+        return(combinedUnsplicedTibble)
     }
-    tableList <- list(
-        "counts.unsplicedSum" = counts.unsplicedSum,
-        "start.unsplicedSum" = start.unsplicedSum,
-        "end.unsplicedSum" = end.unsplicedSum
-    )
-    return(tableList)
+
+
+#' extract new unspliced ranges from readClassSe object for each sample
+#' @importFrom dplyr as_tibble
+#' @importFrom summarizedExperiment rowData rownames
+#' @noRd
+extractNewUnsplicedRanges <- function(readClassSe, sample_id){
+    if (is.character(readClassSe)) 
+        readClassSe <- readRDS(file = readClassSe)
+    rowData <- as_tibble(rowData(readClassSe))
+    pre_names <- rownames(readClassSe)
+    rownames(readClassSe) <- paste0("s",sample_id,"-",pre_names)
+    newUnsplicedSe <- 
+        readClassSe[which(rowData$confidenceType == "unsplicedNew" & 
+                              rowData$readCount > 1)]
```

Comment:
this is because if not creating new object, then rowData(readClassSe) will be repeated three times, I have added another condiiton to filter out equal unspliced read classes